shopinvader_api_address/security/snailmail_letter.xml

```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2023 ACSONE SA/NV
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+<odoo>
+    <!-- allow authenticated partner to snailmail_letter where it is used-->
+    <record id="shopinvader_address_snailmail_letter_rule" model="ir.rule">
+        <field name="name">Shopinvader Address Endpoint rule: snailmail_letter</field>
+        <field name="model_id" ref="snailmail.model_snailmail_letter" />
```

Comment:
To investigate. If this rule stays then shopinvader_api_address must depend on snailmail, which would be... not good.

Reply:
Indeed, I will remove it.

I added this because a check is done in `snailmail` in write of `res.partner` : [https://github.com/odoo/odoo/blob/a4e3662351420b9b87eac386d6bd7b4f075d9fa9/addons/snailmail/models/res_partner.py#LL20C9-L20C9](url)

The problem is that the access to the model `snailmail_letter` is given to only `base.group_user` and `base.group_system`. In our case, we don't want to give any of those group to the endpoint's user (too much rights).

(IMO, an issue can (and will) be created on Odoo regarding the fact that the check should be done in sudo() + I think that the fact that every user can read every `snailmail_letter` is also an issue.. )

To tackle this problem, I think that the best option is to call in sudo mode our functions that does a write on `res.partner` in our services (in the module shopinvader_api_address).